subsys/bluetooth/audio/Kconfig.has

```diff
@@ -0,0 +1,89 @@
+# Bluetooth Audio - Hearing Access Service options
+#
+# Copyright (c) 2022 Codecoup
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if BT_AUDIO
+
+config BT_HAS
+	bool "Hearing Access Service support [EXPERIMENTAL]"
+	select EXPERIMENTAL
+	help
+	  This option enables support for Hearing Access Service.
+
+config BT_HAS_CLIENT
+	bool "Hearing Access Service Client support [EXPERIMENTAL]"
+	select EXPERIMENTAL
+	help
+	  This option enables support for Hearing Access Service Client.
+
+if BT_HAS
+choice BT_HAS_TYPE
+	prompt "Hearing Aid Type selection"
+	help
+	  Select the Hearing Aid Type to compile.
+
+config BT_HAS_HA_TYPE_MONAURAL
+	bool "Monaural Hearing Aid"
+
+config BT_HAS_HA_TYPE_BINAURAL
+	bool "Binaural Hearing Aid"
+
+config BT_HAS_HA_TYPE_BANDED
+	bool "Banded Hearing Aid"
+
+endchoice # BT_HAS_HA_TYPE
+
+config BT_HAS_PRESET_CNT
+	int "Preset record list size"
+	default 4
+	range 0 255
+	help
+	  This option sets the number of Hearing Access Service Presets
+	  that can be registered.
+
+config BT_HAS_IDENTICAL_PRESET_RECORDS
+	bool "Identical preset records in Binaural Hearing Aid Set"
+	depends on BT_HAS_HA_TYPE_BINAURAL
+	help
+	  Set if the list of preset records is identical the list of preset records
+	  on the other member in the Binaural Hearing Aid Set.
+	  This option sets Independent Presets field in Hearing Aid Features to 0b0.
+
+config BT_HAS_PRESET_SYNC_SUPPORT
+	bool "Preset synchronization support"
+	depends on BT_HAS_HA_TYPE_BINAURAL && BT_HAS_IDENTICAL_PRESET_RECORDS
+	help
+	  Set if there is support for active preset changes relay to the other
+	  member in the Binaural Hearing Aid Set.
+
```

Comment:
BT_HAS_WRITABLE_PRESETS is an property that is exposed to the client. As per spec, it tells whether the preset name can be changed by the client. Even if it's disabled, it can be still changed by Server so BT_HAS_PRESET_NAME_DYNAMIC is the server feature to set the name.

Presets that have BT_HAS_WRITABLE_PRESETS property not set can still change it's name when done by server itself. That is how I understand it.

It's just like in GATT, where:
BT_DEVICE_NAME_DYNAMIC shall be set + BT_DEVICE_NAME_GATT_WRITABLE shall be set to allow the GATT client to change the device name

Reply:
Regarding BT_HAS_DYNAMIC_PRESETS I agree, if the support is disabled the bt_has_preset_visibility_set shall return an error. Will fix it.

Reply:
I am not following.
I see your point regarding the PRESET_NAME_DYNAMIC.  But how is the "writable preset support" field decided - is that by seeing if any of the preset names are writable?
And how is the "dynamic presets" field decided?